### PR TITLE
WebGLRenderer: Reset mask before clearing stencil.

### PR DESCRIPTION
**Pull request by OndrejSpanel**
**Description**

When the last object rendered before the `clear` uses `stencilWriteMask`, which may happen when doing multiple passes with such materials, or when using some render targets, the `clear` will not clear bits disabled by the mask.

See https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/clear

> The scissor box, dithering, and buffer writemasks can affect the clear() method

This PR resets mask before doing the clear

### CODE REVIEW

**Comment by OndrejSpanel** [comment]
So many failed screenshots - I am afraid I need to check this more before submitting again.

**Comment by Mugen87** [comment]
Can you please provide a fiddle that demonstrates the error the PR fixes?

**Comment by OndrejSpanel** [comment]
I can try, but it is not trivial. In my case the error was happening randomly in a large scene where stencil was used for object outlining. Depending on object order the error appeared sometimes (old outlines sticking on the screen).

**Comment by OndrejSpanel** [comment]
Here is the issue repro: https://jsfiddle.net/OndrejSpanel/531fa2e8/116/

Steps:

1. Render cyan sphere, writing 4 to stencil
  image

2. Render green sphere with a stencil test value == 4, reseting the write mask to zero (any other render call would do, just reset the write mask):
  image

3. Clear the scene ---- here the issue happens. The stencil is not cleared.

4. Render the green sphere again with a stencil test value == 4 on a freshly cleared backbuffer:
  image
  Nothing should appear, as the stencil buffer should be zero after clear. It is however not cleared and the same shape as in step 4. appears.

File: src/renderers/WebGLRenderer.js [hunk 1]
```diff
@@ -629,7 +629,12 @@ class WebGLRenderer {
 			}
 
 			if ( depth ) bits |= _gl.DEPTH_BUFFER_BIT;
-			if ( stencil ) bits |= _gl.STENCIL_BUFFER_BIT;
+			if ( stencil ) {
+
+				bits |= _gl.STENCIL_BUFFER_BIT;
+				this.state.buffers.stencil.setMask( 0xff );
```
Comment (Mugen87) [inline src/renderers/WebGLRenderer.js:635]: Thanks for the example, that helped to clear things up.

Do you mind using the same default mask value like in `WebGLState` `0xffffffff`? I've tested it and the clear also works with this value.

Comment (OndrejSpanel) [reply src/renderers/WebGLRenderer.js:635]: Makes sense, updated.

**Review by Mugen87 — APPROVED** [review]

**Pull request merged**